### PR TITLE
Fix major issues section 3

### PR DESCRIPTION
**Pull request by svenvanderburg**
Fixes for this lesson (but I referred to the fixes in the respective issues so we can contribute back to the original lesson material):

https://github.com/carpentries-incubator/python-intermediate-development/issues/394
https://github.com/carpentries-incubator/python-intermediate-development/issues/396
https://github.com/carpentries-incubator/python-intermediate-development/issues/367
https://github.com/carpentries-incubator/python-intermediate-development/issues/408

### CODE REVIEW

File: _episodes/34-code-refactoring.md [hunk 1]
```diff
@@ -82,21 +82,25 @@ the tests at all.
 
 > ## Exercise: Write Regression Tests
 > Modify the `analyse_data()` function not to plot a graph and return the data instead.
-> Then, add a new test file called `test_compute_data.py` in the `tests` folder and 
+> Then, adapt the test `test_analyse_data_mock_source.py` in the `tests` folder and
```
Comment (sjvrijn) [inline _episodes/34-code-refactoring.md]: From what I'm reading, the file is (still) called `test_compute_data.py`, with `test_analyse_data_mock_source()` being the test function name, see lines 423 and 431 in the previous file.

**Review by carschno — APPROVED** [review]

**Comment by svenvanderburg** [comment]
@sjvrijn or @carschno oops I added another comment, can you also check that? https://github.com/esciencecenter-digital-skills/python-intermediate-development-gitlab/pull/8/commits/7214c9be44c5143ff8d8184c47472a4b56e5643d

File: _episodes/34-code-refactoring.md (hunk 1 above)
Comment (svenvanderburg) [reply _episodes/34-code-refactoring.md]: Yes, you are right!

**Review by sjvrijn — APPROVED** [review]
LGTM

**Pull request merged**